### PR TITLE
Add support for searching and killing by port - fixes #7

### LINKED ISSUE #7: Support searching and killing by port
See https://github.com/sindresorhus/fkill/commit/11ad78472b07f7788b1fd7140d2734bea3a06c9c and https://github.com/sindresorhus/fkill-cli/pull/33

// @kevva

### PR DESCRIPTION
**Pull request by SamVerschueren**
I gave #7 a go. Looked at how @kevva implemented it in `fkill-cli` and used the same approach. I did not document it yet, but feel free to provide feedback on the implementation details.

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -37,3 +37,13 @@ exports.sorter = (a, b) => {
 
 	return processCompare(a, b);
 };
+
+exports.getPortFromPid = (val, list) => {
```
Comment (SamVerschueren) [inline lib/util.js]: I extracted this method from `fkill-cli` but it seems quite slow. For every process, we loop through the entire PIDs collection. This means that, in algorithm notation, it runs at time complexity `O(n^2)` (as far as my algorithm skills go). Wouldn't it be better if we just swapped the key-value pair from the PIDs map?

Something like this
```js
const swapped = new Map(Array.from(pids.keys()).map(port => [pids.get(port), port])));
```

Then we could just iterate over all the processes and write `swapped.get(process.pid)` which results more in a time complexity of `O(n)` as we don't have a nested loop.

**Comment by SamVerschueren** [comment]
I tried running `alfred-fkill` and `fkill-cli` side by side to see if they return identical results. Most of the time they do, but apparently Chrome is tied to 20 port numbers or so our utils function `getPortFromPid` randomly returns one of the ports. Should we make it more intelligent and filter out the ports first which don't match the search query and then bind the port to the process?

**Comment by SamVerschueren** [comment]
I made the search much better as I'm now able to search for every port tied to one single process.

File: test.js [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+test('search by name', async t => {
 	const alfy = alfyTest();
 	const result = await alfy('bash');
 
 	t.true(result.length > 0);
 });
+
+test('search by port', async t => {
+	const pids = await pidFromPort.list();
+	const ports = [...pids.keys()];
```
Comment (kevva) [inline test.js]: If you only want the first one you could do `const [port] = [...pids.keys()];`.

Comment (SamVerschueren) [reply test.js]: Good point :)! It doesn't seem to work on Travis though so might have to remove it. Or disable it on CI.

Comment (kevva) [reply test.js]: Maybe create a fixture with a port and try searching for that? Could be that `ports` here is empty on Travis. Like we do here https://github.com/sindresorhus/fkill/blob/master/test.js#L87-L93.

Comment (SamVerschueren) [reply test.js]: Oh yes! Great idea!

**Review by kevva — APPROVED** [review]

**Comment by kevva** [comment]
Looking good!

**Comment by sindresorhus** [comment]
Looking good!

**Comment by SamVerschueren** [comment]
Thanks for the feedback guys! Much appreciated! Released it!

**Pull request merged**